### PR TITLE
tests: install docker after lxd

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Test to install docker after LXD.

### CODE REVIEW

**Review by lengau — APPROVED** [review]
As you pointed out in Mattermost this isn't great in terms of solving the underlying problem, but it is a decent workaround.

Given that Fedora 36 is EOL as of Tuesday and Ubuntu 18.04 enters ESM this month, maybe it's okay for us to drop these? (I'd want to add at least a replacement for fedora 36 though. ) I'm giving Fedora 37, Fedora CoreOS, Debian (sid & bullseye) and Rocky Linux (8 & 9) a go  and will report back

File: spread.yaml [hunk 1]
```diff
@@ -84,6 +78,11 @@ prepare: |
   lxd waitready --timeout=30
   lxd init --auto
 
+  snap install docker
+
+  # make sure docker is working
```
Comment (tigarmo) [inline spread.yaml:83]: possibly add a link to the mattermost thread here? To help us in the future

**Comment by lengau** [comment]
So fun fact: The following other distros fail the tests in the same way if Docker gets installed before LXD:

* Debian (bullseye and sid)
* Fedora 37
* CentOS 7

**Comment by mr-cal** [comment]
> So fun fact: The following other distros fail the tests in the same way if Docker gets installed before LXD:
> 
>     * Debian (bullseye and sid)
> 
>     * Fedora 37
> 
>     * CentOS 7

I bet it is because they do not have `nftables` or `iptables-nft` installed.

**Comment by mr-cal** [comment]
Latest failures: 

https://github.com/canonical/rockcraft/actions/runs/5008322142/jobs/8990735668?pr=262#step:5:126 is a known lxd bug: https://github.com/lxc/lxd/issues/11422 that occurs when the CPU is at 100% utilization

I'm not sure about the other failure (https://github.com/canonical/rockcraft/actions/runs/5008322142/jobs/8990735668?pr=262#step:5:359).

**Comment by tigarmo** [comment]
That second failure is a race condition I think; the output is generated but not before "docker logs" is run.
I'm prototyping a fix here

**Comment by tigarmo** [comment]
actually @mr-cal @sergiusens can either of you bypass-merge this? I'll do the fix in the flaky spread test in a separate PR

**Pull request merged**